### DISCUSSION TITLE
Lazy evaluation may lead to unexpected results

### DISCUSSION THREAD

**Discussion by BerriJ** [rstudio/gt#877] [General]
Hi, I recently discovered an interesting (in hindsight) problem. Consider you want to make a simple 3x3 table containing NAs. Now you want to add 1,2,3 under the 3 respective diagonal elements of the table.

What I tried: 

```
library(dplyr)
library(gt)

df <- as.data.frame(matrix(NA, nrow = 3, ncol = 3))
tab <- gt(df)
for (i in 1:3) {
    tab <- tab %>%
        text_transform(locations = cells_body(
            columns = all_of(i),
            rows = all_of(i)
        ), fn = function(x) {
            paste0(x, "<br>", i)
        })
}
tab See result here: [tab](https://i.stack.imgur.com/0c1fd.png)
```

However, add "3" to all diagonal elements of the table. This [SO Thread ](https://stackoverflow.com/questions/70959060/how-to-use-gttext-transform-inside-a-loop/70960002#70960002) suggests points out that R's lazy evaluation is messing things up here. The SO Thread also contains a solution > using `lapply`. 

So there are some problems I see:

- People that are aware of the problem have to use `lapply` or apply or something. But translating loops into `lapply` calls is not always easy, especially if you deal with nested loops.
- People that are not aware won't know why they get unexpected results

I think looping through (part of) a table is a relatively common scenario. Is there something we can do about this? So for example can we force non-lazy evaluation inside of for-loops?

Let's Discuss! :)

**Comment by rich-iannone** [comment]
I've seen a similar looping issue before and I totally agree that this is not an uncommon thing to want to do. I'll have a closer look at this and I suspect it's a gt problem. Will create an issue from this.
